src/transformers/trainer.py

```diff
@@ -522,14 +522,17 @@ def __init__(
             getattr(model, "hf_quantizer", None) is not None and model.hf_quantizer.is_trainable
         )
 
+        _is_model_quantized_and_trainable = getattr(model, "is_quantized", False) and getattr(
+            model, "is_compressed_tensors_quantized", False
+        )
```

Comment:
To check if the model is quantized, let's use `getattr(model, "hf_quantizer", None) is not None` instead. Then you can also get the quantization method from model.hf_quantizer. If you are doing qat, we can think about creating another property `is_qat_trainable`.